onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -364,6 +367,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
     compute_info.compute_func = [](FunctionState state, ONNXRunTimeTensor* input_tensors, size_t num_inputs, ONNXRunTimeTensor* output_tensors, size_t num_outputs) {
       ORT_UNUSED_PARAMETER(num_inputs);
       ORT_UNUSED_PARAMETER(num_outputs);
+      std::lock_guard<OrtMutex> glock(tensorrt_mu);
```

Comment:
is the mutex really needed around all of the compute func?
or just when it runs the TRT inference, i.e. the context->enqueue() ?